src/web/src/components/CourseList.vue

```diff
@@ -128,27 +149,41 @@ export default {
       this.selectedSubsemester = options[0].value;
       return options;
     },
-    /**
-     * Returns a list of courses that match the selected filters
-     * @returns {Course[]}
-     */
-    filteredCourses() {
-      return this.courses.filter(
-        ({ date_start, date_end, department, title, semester }) => {
-          return (
-            (!this.selectedSubsemester ||
-              (this.selectedSubsemester.date_start.getTime() ===
-                date_start.getTime() &&
-                this.selectedSubsemester.date_end.getTime() ===
-                  date_end.getTime())) &&
-            (!this.selectedDepartment ||
-              this.selectedDepartment === department) &&
-            (!this.textSearch ||
-              title.includes(this.textSearch.toUpperCase())) &&
-            this.selectedSemester === semester
-          );
-        }
+    // return a list of the titles of each course.
+    mapCourseNames() {
+      return this.courseList
+        .filter(
+          (course) =>
+            this.selectedDepartment === null ||
+            course.department === this.selectedDepartment
+        )
+        .map((course) => {
+          if (course.full_title !== null) return course.full_title;
```

Comment:
can take advantage of javascript values being truthy and do `course.full_title || course.title`